### PR TITLE
Add expand symint python test and re-enable some previously disabled dynamic shape tests.

### PR DESCRIPTION
**Pull request by vanbasten23**
This PR:
- add expand_symint test to replace the cpp counterpart test: AtenXlaTensorTest.TestExpandSymIntSymbolic and AtenXlaTensorTest.TestExpandSymIntDynamic
- re-enable test AtenXlaTensorTest.TestNonzero
- re-enable test TestDynamicShape.test_nonzero_cast

### CODE REVIEW

**Review by alanwaketan — APPROVED** [review]
LGTM.

File: test/test_operations.py [hunk 1]
```diff
@@ -760,6 +757,23 @@ def test_nonzero_cast(self):
     t2 = torch.nonzero(t1.int()).float()
     xm.mark_step()
 
+  def test_expand_symint_correctness(self):
```
[the remark this answers is not in the document]
Comment (vanbasten23) [reply test/test_operations.py]: done

File: test/test_dynamic_shapes.py [hunk 2]
```diff
@@ -139,6 +140,44 @@ def test_empty_symint(self):
...
+    dev = xm.xla_device()
+    size1 = 5
+    size2 = 2
+    t1 = torch.ones([size1, size2])
+    expand_out_aten = t1.expand(2, size1, size2)
+
+    t2 = torch.zeros([size1, size2], device=dev)
+    t2[3][0] = 1
+    t2[3][1] = 1
+    # t2 has size [<=10, 2]
```
Comment (JackCaoG) [inline test/test_dynamic_shapes.py:174]: do you mean t3? t2 is a static tensor.

File: test/test_dynamic_shapes.py [hunk 3]
```diff
@@ -139,6 +140,44 @@ def test_empty_symint(self):
...
+    t3 = torch.nonzero(t2)
+    t4 = torch.ones([size1, size2], device=dev)
+    expand_out_xla = t4.expand(t3.shape[0], size1, size2)
```
Comment (JackCaoG) [inline test/test_dynamic_shapes.py:177]: I am a bit confuse why `t4.expand` takes 3 arguments..

File: test/test_dynamic_shapes.py (hunk 2 above)
Comment (vanbasten23) [reply test/test_dynamic_shapes.py:174]: right, my bad. Do you mind if I update the comment in a following pr?

File: test/test_dynamic_shapes.py (hunk 3 above)
Comment (vanbasten23) [reply test/test_dynamic_shapes.py:177]: per https://pytorch.org/docs/stable/generated/torch.Tensor.expand.html, it's just a size (t3.shape[0], size1, size2). I could also make it a single argument such as `t4 = t3.expand((t3.shape[0], size1, size2))`. The test `test_simple_expand_on_2d_tensor` in this files provides a few other variants.

**Review by JackCaoG — APPROVED** [review]

**Pull request merged**